src/content/docs/en/guides/integrations-guide/react.mdx

```diff
@@ -114,6 +114,54 @@ To use your first React component in Astro, head to our [UI framework documentat
 * 💧 client-side hydration options, and
 * 🤝 opportunities to mix and nest frameworks together
 
+## Integrate Actions with `useActionState()` (experimental)
+
+[Astro Actions][astro-actions] can be integrated with [React's `useActionState()` hook](https://react.dev/reference/react/useActionState) to update client-side state when triggering actions during form submission.
```

Comment:
```suggestion
The `@astrojs/react` integration provides two experimental functions for use with [Astro Actions][astro-actions]: `experimental_withState()` and `experimental_getActionState()`.

These can be used with [React's `useActionState()` hook](https://react.dev/reference/react/useActionState) to update and read client-side state when triggering actions during form submission.
```
What about an introduction that front-loads the things we're documenting, and makes it clear that these are to be used with Actions? Then, we describe what they do.

Reply:
Looks good to me, maybe I would be a bit more explicit with the second sentence to be clear that these functions are only to be used with `useActionState()` and don't make sense otherwise. Also, changing the order a bit in the end
```suggestion
The `@astrojs/react` integration provides two experimental functions for use with [Astro Actions][astro-actions]: `experimental_withState()` and `experimental_getActionState()`.

These are to be used alongside [React's `useActionState()` hook](https://react.dev/reference/react/useActionState), allowing to read and update client-side state when triggering actions during form submission.
```

Reply:
OK, agree about the emphasis on `useActionState`! Can we be more brutal about the verbosity, and get away with this:


> These are used with [React's `useActionState()` hook](https://react.dev/reference/react/useActionState) to read and update client-side state when triggering actions during form submission.